query/stdlib/influxdata/influxdb/operators.go

```diff
@@ -117,6 +118,31 @@ func (s *ReadRangePhysSpec) TimeBounds(predecessorBounds *plan.Bounds) *plan.Bou
 	}
 }
 
+type Window struct {
```

Comment:
My first version was just `Every int64`. But later I felt I'm not 100% convinced that we should leave out the overlapping case following the discussion here https://github.com/influxdata/idpe/pull/6703/files#r405818778.

@nathanielc thinks window with `offset` and `period` makes the workload not stable and predictable, but I don't see why. I will continue the discussion there.

Reply:
I think we could capture them into a single value. There's no reason why we can't have:

```
WindowEvery int64
WindowPeriod int64
```

If you want it to be a separate struct though, please export all of the types so this should be:

```
type Window struct {
    Every int64
}
```

I'm a bit skeptical this is better though since if we reuse this struct somewhere else we may not implement it. I would prefer if it was directly tied into the physical spec that is implemented to avoid a scenario where we reuse this struct later, add `Period int64`, then forget to implement `Period` in the second physical spec.